tests/testthat/test-type-date-time.R

```diff
@@ -173,21 +175,24 @@ test_that("safe casts work as expected", {
   datetime_l <- as.POSIXlt("1970-02-01", tz = "UTC")
 
   expect_equal(vec_cast(NULL, datetime_c), NULL)
-  expect_equal(vec_cast(2678400, datetime_c), datetime_c)
-  expect_equal(vec_cast("1970-02-01", datetime_c), datetime_c)
   expect_equal(vec_cast(datetime_c, datetime_c), datetime_c)
   expect_equal(vec_cast(datetime_l, datetime_c), datetime_c)
   expect_equal(vec_cast(as.Date(datetime_c), datetime_c), datetime_c)
-  expect_equal(vec_cast(list(datetime_c), datetime_c), datetime_c)
 
   expect_equal(vec_cast(NULL, datetime_l), NULL)
-  expect_equal(vec_cast(2678400, datetime_l), datetime_l)
-  expect_equal(vec_cast("1970-02-01", datetime_l), datetime_l)
   expect_equal(vec_cast(datetime_c, datetime_l), datetime_l)
   expect_equal(vec_cast(datetime_l, datetime_l), datetime_l)
   expect_equal(vec_cast(as.Date(datetime_l), datetime_l), datetime_l)
-  expect_equal(vec_cast(list(datetime_l), datetime_l), datetime_l)
   expect_error(vec_cast(raw(), datetime_l), class = "vctrs_error_incompatible_cast")
+
+  # These used to be allowed
+  expect_error(vec_cast(2678400, datetime_c), class = "vctrs_error_incompatible_cast")
+  expect_error(vec_cast("1970-02-01", datetime_c), class = "vctrs_error_incompatible_cast")
```

Comment:
This change makes me slightly nervous because `x < "2000-01-01"` is a common operation. Maybe we should allow ISO8601 dates-times in strings? That feels a bit similar to allowing "double" 1 to go into a integer.

Reply:
This doesn't work with CRAN vctrs because dates and characters don't have a common type. I think it'd make sense to add a common type, and only allow casting towards dates. I have opened an issue: https://github.com/r-lib/vctrs/issues/967